### PR TITLE
Refactor conditional resolution framework into conditional satisfiability evaluation framework

### PR DESCRIPTION
**Pull request by custode**
Refactor the models and methods related to the conditional resolution framework to express the intent of the framework better.

The original naming convention could be confusing as the conditional dependency graph can be inferred as the graph with the edge pointing from the **object with the dependency to the object which is depended upon**, which is the way we represented the dependency graph in the database. It is renamed to user satisfiability graph to differentiate itself from the representation in the database as the edge now points in the opposite direction, **from the object which is depended upon to the object with the dependency**.

Also, **resolution is replaced with evaluation** as the purpose of the framework is to evaluate the conditionals that are currently satisfied by the user.

### CODE REVIEW

**Comment by lowjoel** [comment]
so we "build the satisfiability graph" but we "evaluate the satisfiability graph for a course user", correct?

File: app/models/course/condition.rb [hunk 1]
```diff
@@ -20,7 +20,9 @@ class << self
     # @return [Object] acts_as_conditionals objects belonging to the given course
     def conditionals_for(course)
       dependent_class_to_condition_class_mapping.keys.map do |conditional_name|
-        conditional_name.constantize.where(course_id: course)
+        if conditional_name.constantize.include? ActiveRecord::Base::ConditionalInstanceMethods
```
Comment (lowjoel) [inline app/models/course/condition.rb]: `next unless conditional_name.constantize.is_a?(ActiveRecord::Base::ConditionalInstanceMethods)`

**Comment by lowjoel** [comment]
Hopefully you really just renamed in the second commit, there's a lot of stuff changed.

**Comment by custode** [comment]
You are right! We build the satisfiability graph, and make use of the built satisfiability graph to evaluate the satisfiability of the conditionals for a course user.

**Comment by custode** [comment]
Yes the second commit is purely renaming.

Dependency graph -> Satisfiability graph
Resolve -> Evaluate

File: app/models/course/condition.rb (hunk 1 above)
Comment (custode) [inline app/models/course/condition.rb]: is_a? seems to be working on the instance of the class only. Will continue to use include? for this part unless you have other alternative.

Comment (lowjoel) [inline app/models/course/condition.rb]: sorry, `kind_of?`

Comment (lowjoel) [inline app/models/course/condition.rb]: wait that doesn't work either. nevermind. parenthesize the include?() call.

**Pull request merged**